# Data Mining Project - Speed Dating Experiment Analysis
 
**Authors:** Tomasz Kisielewski, Robert Gosz, Antoni Kiciński  
 **Dataset:** [Speed Dating Experiment](https://www.kaggle.com/datasets/annavictoria/speed-dating-experiment)

----

In [2]:
import kagglehub

# Download latest version
path = kagglehub.dataset_download("annavictoria/speed-dating-experiment")

print("Path to dataset files:", path)

100%|██████████| 374k/374k [00:00<00:00, 796kB/s]

Extracting files...
Path to dataset files: C:\Users\robert\.cache\kagglehub\datasets\annavictoria\speed-dating-experiment\versions\1


In [18]:
import pandas as pd

df = pd.read_csv(f"{path}/Speed Dating Data.csv", encoding='latin1')
print(df.head())
print("\nShape of the dataset:")
print(df.shape)

   iid   id  gender  idg  condtn  wave  round  position  positin1  order  ...  \
0    1  1.0       0    1       1     1     10         7       NaN      4  ...   
1    1  1.0       0    1       1     1     10         7       NaN      3  ...   
2    1  1.0       0    1       1     1     10         7       NaN     10  ...   
3    1  1.0       0    1       1     1     10         7       NaN      5  ...   
4    1  1.0       0    1       1     1     10         7       NaN      7  ...   

   attr3_3  sinc3_3  intel3_3  fun3_3  amb3_3  attr5_3  sinc5_3  intel5_3  \
0      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
1      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
2      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
3      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
4      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   

   fun5_3  amb5_3  
0     NaN     NaN  
1     NaN 

## 1. General Description of the Dataset
 
The Speed Dating Experiment dataset was collected between 2002 and 2004 at Columbia University as part of a research study on partner preferences and romantic decision-making. The experiment consisted of 21 sessions (waves) of speed dating events, in which participants engaged in a series of 4-minute one-on-one conversations with potential partners. After each date, both participants independently indicated whether they would like to meet the other person again. A match occurred when both participants answered "yes."
 
The dataset covers much more than just the dating decisions. Before the event, participants were asked what they look for in a partner - they split 100 points across six traits: attractiveness, intelligence, sincerity, fun, ambition, and shared interests. They also rated themselves on those same traits. This makes it possible to compare what people say they want with how they actually make decisions. After each date, participants rated their partner on the same set of traits, gave an overall liking score, and estimated whether the partner would also say "yes." Basic personal information was also collected, such as age, field of study, career plans, and hobbies.
 
The dataset contains **8,378 records** representing individual speed dates, involving **551 unique participants** across **21 waves**. Each row corresponds to one speed date from the perspective of one participant. Since each date is recorded separately from both participants' perspectives, every date appears as two rows in the dataset.
 
---

## 2. Data Mining Goals and Success Criteria
 
### Primary Goal
**Classification:** Predict whether a speed date will result in a mutual match (`match = 1`), meaning both participants said "yes" to meeting again. The goal is not to predict long-term romantic success, but to understand what drives mutual attraction during a 4-minute speed date - specifically, which participant ratings, stated preferences, and personal characteristics make it more likely that both people will say "yes."
 
### Supporting Analytical Goals
The following exploratory analyses will support the primary classification goal and provide actionable insights:
 
1. **Baseline match rate** - Establish what proportion of speed dates result in a mutual match.
2. **Gender differences** - Determine whether men or women are more likely to say "yes" during speed dates.
3. **Feature importance for match** - Identify which attributes (attractiveness, fun, intelligence, sincerity, ambition, shared interests, overall liking, perceived reciprocity, and age) are most predictive of a match.
4. **Stated vs. revealed preferences** - Investigate whether participants' self-declared preferences (what they say they value in a partner) align with the attributes that actually predict their decisions.
5. **Self-assessment vs. outcomes** - Examine whether participants who rate themselves higher receive more matches.
6. **Pickiness effect** - Determine whether being more selective (saying "yes" to fewer people) leads to better or worse match outcomes.

### Success Criteria
- **Classification model:** The classification model should achieve at least 70% accuracy in predicting whether a speed date results in a mutual match. 
- **Hidden patterns:** The exploratory analysis should uncover non-obvious relationships in the data - for example, whether being more selective actually increases your chances of a match, or whether what people say they want in a partner differs from what they actually respond to.
- **Golden client:** Based on the analysis, we aim to identify a profile of a participant most likely to result in a match - a combination of traits and behaviors (e.g. high fun rating, similar age, shared interests) that consistently predict mutual attraction.
---

## 3. Dataset Characteristics
 


In [9]:
print("Dataset Characteristics:\n")
print(f"Number of records:       {len(df)}")
print(f"Number of attributes:    {len(df.columns)}")
print(f"Unique participants:      {df['iid'].nunique()}")
print(f"Number of waves:         {df['wave'].nunique()}")
match_rate = df['match'].sum() / len(df)
print(f"Match rate (positive):   {match_rate:.2%}")
print(f"No match rate (negative):{1 - match_rate:.2%}")
print(f"Total matches:           {df['match'].sum()}")
print("Match values:")
print(df['match'].value_counts())

Dataset Characteristics:

Number of records:       8378
Number of attributes:    195
Unique participants:      551
Number of waves:         21
Match rate (positive):   16.47%
No match rate (negative):83.53%
Total matches:           1380
Match values:
match
0    6998
1    1380
Name: count, dtype: int64


| Property | Value |
|---|---|
| Origin | Columbia University speed dating experiment (2002–2004) |
| Format | Single CSV file |
| Number of records | 8,378 |
| Number of attributes | 195 |
| Unique participants | 551 |
| Number of waves (sessions) | 21 |
| Level of observation | One row = one speed date from one participant's perspective |
| Target variable | `match` (binary: 0 or 1) |
| Class balance | ~16.5% positive (match), ~83.5% negative |
 
The dataset consists of a single flat file. Because each date is recorded twice (once per participant), pairs of rows share the same date but represent different viewpoints.
 
---